Updated summary geneartion with .json input

stefherra opened the pull request:
I updated the python script so it matches python3 standards, I thought this may be good and it was only altering the print statements. I then made sure that the file that is being read is the new .json file, not the old .csv. 

I added the automatic generation of a .csv file so the need of copy-pasting the printed outcome to an excel file is not necessary anymore. The results are still printed so the old way should still work. 

There are some small changes in the outcome by copy-pasting and automatically generated file because of the rounding of the numbers.

lynxlynxlynx commented on generate-summary.py:
```diff
@@ -11,10 +13,11 @@
-reader = csv.reader(file)
-for line in reader:
-	record = line[14]
+file = open(os.getcwd()+"\\assets\\data\\komunale.json", "rb")
```
Why not just use a relative path like before? Would be much cleaner ...

stefherra replied:
I did it this way because "komunale.json" is not located in the same folder as the python script in the original directory, but it's located under assets > data. We could also change it back to the relative path if that would be better, as I agree it is much cleaner

lynxlynxlynx replied:
Well yeah, "assets/data/komunale.json" is still a relative path.

lynxlynxlynx commented on generate-summary.py:
```diff
@@ -1,7 +1,9 @@
 import csv
-import sys # exceptions
+import json # needed for json import
+import os # needed to find working directory
```
These two comments are redundant, but they don't hurt either.

stefherra replied:
I agree, could remove these

lynxlynxlynx commented on generate-summary.py:
```diff
@@ -41,14 +41,23 @@
+		out2 = []
 		for c in t[1]:
 			out+= str(float(c*100) / float(cnt))[:4] + " " 
+			out2.append(round(((c*100)/cnt),1))
```
Like before, this doesn't guarantee the shares to add up to 100%, but it's fine, we don't need to care about promiles.

lynxlynxlynx commented on assets/data/cat-shares-new.csv:
```diff
@@ -0,0 +1,156 @@
+0.0,0.0,0.0,0.0,0.0,0.0,23.1,0.0,76.9
```
this looks fine, so you can just overwrite the existing file.  Which means you can also change the filename in the writer call.

lynxlynxlynx commented on generate-summary.py:
```diff
@@ -41,14 +41,23 @@
 		print(t[0], t[1], cnt)
+
+# Save results to csv file 
+with open(os.getcwd()+"\\assets\\data\\cat-shares-new.csv", "w", newline='') as f:
```
Windows understands forward slashes, so in the future just use `/` as path separators in python code — it'll be cleaner.

lynxlynxlynx commented:
Thanks, you grasped git well so far! Let me know if any of the comments were not clear enough.

stefherra commented:
Hi, you're comments were clear! Git is a nice way to work like this. 

I took your comments into account and removed the redundant comments, readjusted the opening of the file to the relative path, used <code>/</code> in the writer path and changed the file that is written to to the original! Let me know what you think.

lynxlynxlynx commented on generate-summary.py:
```diff
@@ -13,7 +13,7 @@
 	total[i][0] = i+1
 
-file = open(os.getcwd()+"\\assets\\data\\komunale.json", "rb")
+file = open("komunale.json", "rb")
```
Now the path is wrong — just prepend it like I mentioned in the first comment today.

lynxlynxlynx commented on generate-summary.py:
```diff
@@ -58,6 +58,6 @@
 
 # Save results to csv file 
-with open(os.getcwd()+"\\assets\\data\\cat-shares-new.csv", "w", newline='') as f:
+with open(os.getcwd()+"/assets/data/cat-shares.csv", "w", newline='') as f:
```
You can simplify this path logic here as well.

lynxlynxlynx commented on komunale.json:
```diff
@@ -0,0 +1,274 @@
+{
```
nonono :)

lynxlynxlynx replied:
By fixing the code there will be no need to break the nice file organization.

stefherra commented:
Ahh, I had no I idea that you could use a path that way. This makes life so much easier in so many cases. I deleted the .json file again from the main folder and it can only be found under assets>data again. Also, I think paths should be fixed now.

lynxlynxlynx commented on generate-summary.py:
```diff
@@ -26,26 +29,35 @@
+		result.append(out2)
 	else:
-		print t[0], t[1], cnt
+		print(t[0], t[1], cnt)
```
This branch is missing addition to the file writing list.

stefherra replied:
would I be correct to say that if <code>rel=False</code>, then we don't want the percentages but the absolute numbers? In this case, do we want the cat-shares.csv file to be overwritten with absolute numbers or a separate file to be created?

lynxlynxlynx replied:
Yes and the same file is fine, since it is trivial to regenerate. This absolute mode is only used rarely, when there's a need for some base statistics.

lynxlynxlynx commented:
Other than that and a slight redundancy in having two different ways to do the out2 block, it looks fine. 👍🏾

stefherra commented:
I added writing the absolute numbers to the file. In the print, the number and sum of is also printed, but I did not include this in the file that is written, as I thought it was more logical to just write the absolute numbers.

lynxlynxlynx commented:
Thanks!
